src/mesh/test/tstX3D_Draco_Mesh_Reader.cc

```diff
@@ -208,13 +208,56 @@ void build_x3d_mesh_2d(rtt_c4::ParallelUnitTest &ut) {
   return;
 }
 
+//----------------------------------------------------------------------------------------//
+// Parse a larger Voronoi X3D file format and compare to reference
+void read_voronoi_mesh(rtt_c4::ParallelUnitTest &ut) {
+
+  // >>> PARSE MESH
+
+  const std::string inputpath = ut.getTestSourcePath();
+  const std::string filename = inputpath + "pipe_voronoi.x3d.in";
+  const std::vector<std::string> bdy_filenames;
+  const std::vector<unsigned> bdy_flags;
+
+  // construct reader
+  std::shared_ptr<X3D_Draco_Mesh_Reader> x3d_reader(
+      new X3D_Draco_Mesh_Reader(filename, bdy_filenames, bdy_flags));
+
+  // read mesh
+  x3d_reader->read_mesh();
+
+  // >>> CHECK HEADER DATA
+
+  FAIL_IF_NOT(x3d_reader->get_process() == 0);
+  FAIL_IF_NOT(x3d_reader->get_numdim() == 2);
+  FAIL_IF_NOT(x3d_reader->get_numcells() == 1590);
+  FAIL_IF_NOT(x3d_reader->get_numnodes() == 2291);
+
+  // >>> CHECK CELL-NODE DATA
+
+  FAIL_IF_NOT(x3d_reader->get_celltype(0) == 5);
+
+  std::vector<unsigned> test_cellnodes = {0,   938, 938, 212, 212,
+                                          199, 199, 939, 939, 0};
+  FAIL_IF_NOT(x3d_reader->get_cellnodes(0) == test_cellnodes);
+
+  // CHECK MATIDS
+  FAIL_IF_NOT(x3d_reader->get_matid(0) == "1");
+
+  // successful test output
+  if (ut.numFails == 0)
+    PASSMSG("2D X3D_Draco_Mesh_Reader parsing tests ok.");
+  return;
+}
+
 //----------------------------------------------------------------------------//
 int main(int argc, char *argv[]) {
   rtt_c4::ParallelUnitTest ut(argc, argv, rtt_dsxx::release);
   try {
     Insist(rtt_c4::nodes() == 1, "This test only uses 1 PE.");
     read_x3d_mesh_2d(ut);
     build_x3d_mesh_2d(ut);
+    read_voronoi_mesh(ut);
```

Comment:
I left a note in the general comments requesting that `draco/.codecov.yml` be modified so that the new input file is ignored by the coverage tool.